### ISSUE TITLE
🔒 Saving lockfile... error: failed to open lockfile: EBUSY

### ISSUE THREAD

**Issue by JiriMachynka** [oven-sh/bun#9943]
### What version of Bun is running?

1.1.1+ca1dbb4eb

### What platform is your computer?

Microsoft Windows NT 10.0.22631.0 x64

### What steps can reproduce the bug?

`bun i` inside a project

### What is the expected behavior?

Bun should install dependencies correctly and properly update its lockfile.

### What do you see instead?

![bun-lockfile-error](https://github.com/oven-sh/bun/assets/65855539/f6c5adfe-2416-45a9-b15e-587f82f57b74)

**Comment by consp1racy** [comment]
also `bun pm migrate`

**Comment by Jarred-Sumner** [comment]
Bun v1.1.2 will ship in less than 12 hours and will fix this

**Comment by Jarred-Sumner** [comment]
The fix is merged

In about an hour, feel free to run `bun upgrade --canary` and give the fix a try

**Comment by SKumarSpace** [comment]
Hello, I ran the `bun upgrade --canary` command and it panicked. 

```sh
$ bun upgrade --canary

uh-oh: attempt to unwrap error
bun will crash now 😭😭😭

----- bun meta -----
Bun v1.1.1 (ca1dbb4e) Windows x64
UpgradeCommand:
Elapsed: 1806ms | User: 0ms | Sys: 0ms
RSS: 95.79MB | Peak: 119.14MB | Commit: 0.15GB | Faults: 35545
----- bun meta -----

0   00007FF77B3BED58
1   ???
2   ???
3   ???
4   ???
5   ???
6   ???
7   ???
8   ???
9   ???

Search GitHub issues https://bun.sh/issues or join in #windows channel in https://bun.sh/discord

thread 116772 panic: attempt to unwrap error
???:?:?: 0x7ff779ffbdd1 in ??? (bun.exe)
???:?:?: 0x7ff779ff9945 in ??? (bun.exe)
???:?:?: 0x7ff779ffc406 in ??? (bun.exe)
???:?:?: 0x7ff77a0090e7 in ??? (bun.exe)
???:?:?: 0x7ff779d31603 in ??? (bun.exe)
???:?:?: 0x7ff77c8b3ec3 in ??? (bun.exe)
???:?:?: 0x7ffc7daf257c in ??? (KERNEL32.DLL)
???:?:?: 0x7ffc7fe0aa47 in ??? (ntdll.dll)
```

**Comment by lokimckay** [comment]
Got the same panic error as SkumarSpace
Reinstalling bun fixed this issue for me
Windows 10 10.0.19045
Bun 1.1.2 after reinstall

**Comment by JohnC0de** [comment]
For anyone comming from web search(like me):

Dylan fixed it on #13563 and will be included in bun v1.1.27
You can try the fix now with canary (bun upgrade --canary)

Now on build you can run something with bun and also download deps, example
Terminal 1: bun dev
Terminal 2: bunx shadcn init
